code/game/data_huds.dm

```diff
@@ -0,0 +1,258 @@
+ /* Data HUDs have been rewritten in a more generic way.
+ * In short, they now use an observer-listener pattern.
+ * See code/datum/hud.dm for the generic hud datum.
+ * Update the HUD icons when needed with the appropriate hook. (see below)
+ */
+
+/* DATA HUD DATUMS */
+
+/atom/proc/add_to_all_human_data_huds()
+	for(var/datum/atom_hud/data/human/hud in huds) hud.add_to_hud(src)
+
+/atom/proc/remove_from_all_data_huds()
+	for(var/datum/atom_hud/data/hud in huds) hud.remove_from_hud(src)
+
+/datum/atom_hud/data
+
+/datum/atom_hud/data/human/medical
+	hud_icons = list(HEALTH_HUD, STATUS_HUD)
+
+/datum/atom_hud/data/human/medical/basic
+
+/datum/atom_hud/data/human/medical/basic/proc/check_sensors(mob/living/carbon/human/H)
+	if(!istype(H)) return 0
+	var/obj/item/clothing/under/U = H.w_uniform
+	if(!istype(U)) return 0
+	if(U.sensor_mode <= 2) return 0
+	return 1
+
+/datum/atom_hud/data/human/medical/basic/add_to_single_hud(mob/M, mob/living/carbon/H)
+	if(istype(M,/mob/dead/observer))
+		return 1
+	if(check_sensors(H))
+		..()
+
+/datum/atom_hud/data/human/medical/basic/proc/update_suit_sensors(mob/living/carbon/H)
+	check_sensors(H) ? add_to_hud(H) : remove_from_hud(H)
+
+/datum/atom_hud/data/human/medical/advanced
+
+/datum/atom_hud/data/human/security
+
+/datum/atom_hud/data/human/security/basic
+	hud_icons = list(ID_HUD)
+
+/datum/atom_hud/data/human/security/advanced
+	hud_icons = list(ID_HUD, IMPTRACK_HUD, IMPLOYAL_HUD, IMPCHEM_HUD, WANTED_HUD)
+
+/datum/atom_hud/data/diagnostic
+	hud_icons = list (DIAG_HUD, DIAG_STAT_HUD, DIAG_BATT_HUD, DIAG_MECH_HUD)
```

Comment:
Were the HUD's added prior to them being simple animals or after? IF they were added prior, then you'll just have to dig through some of their older code (the machines->simple animals change was recent).